Subject: [PATCH] Keep access point list updated once in short time.

The function onAccessPointsChanged will be called many times and
access points will be updated many times in 300ms.
Remove the old runnable to keep access points updating only once
in 300ms.

diff --git a/src/com/android/settings/wifi/WifiSettings.java b/src/com/android/settings/wifi/WifiSettings.java
index 190af4d2e4..9a77de4135 100644
--- a/src/com/android/settings/wifi/WifiSettings.java
+++ b/src/com/android/settings/wifi/WifiSettings.java
@@ -703,6 +703,10 @@ public void onAccessPointsChanged() {
     private void updateAccessPointsDelayed() {
         // Safeguard from some delayed event handling
         if (getActivity() != null && !mIsRestricted && mWifiManager.isWifiEnabled()) {
+            // The function onAccessPointsChanged will be called many times and access points
+            // will be updated many times in 300ms.
+            // Remove the old runnable to keep access points updating only once in 300ms.
+            getView().removeCallbacks(mUpdateAccessPointsRunnable);
             setProgressBarVisible(true);
             getView().postDelayed(mUpdateAccessPointsRunnable, 300 /* delay milliseconds */);
         }
